hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/AbstractEvictableRecordStore.java

```diff
@@ -213,42 +212,23 @@ private void cleanUp(long now) {
         if (size() == 0) {
             return;
         }
+
         if (shouldEvict(now)) {
-            evict();
-            lastEvictionTime = now;
-            readCountBeforeCleanUp = 0;
+            int sizeToEvict = evictor.findSizeToEvict(this);
```

Comment:
This is an unclear API like I suggested above. You expose less of your implementation detail if you can say to the evictor.evict().

The evictor will keep track of stuff like lastEvictionTime, readCountBeforeCleanup etc etc. 

This way the Evictor becomes a very 'simple' abstraction with potentially complex behavior. Currently it is a bit smeared over multiple classes.